src/Cardano/Address/Style/Shelley.hs

```diff
@@ -255,12 +261,16 @@ instance HasNetworkDiscriminant Shelley where
 
 -- | Error reported from trying to create a network discriminant from number
 --
--- @since 1.0.0
+-- @since 1.0.1
 newtype MkNetworkDiscriminantError
     = ErrWrongNetworkTag Word8
       -- ^ Wrong network tag.
     deriving (Eq, Show)
 
+-- | Construct 'NetworkDiscriminant' for Cardano 'Shelley' from number.
+-- If number is invalid `MkNetworkDiscriminantError
+--
+-- @since 1.0.1
 mkNetworkDiscriminant
     :: Word8
     -> Either MkNetworkDiscriminantError (NetworkDiscriminant Shelley)
```

Comment:
We need to make sure the documentation mentions the invariant and tells users that we expect a number between 0 and 15. Otherwise, it's kinda cryptic.

Reply:
done